### PR TITLE
Add `apache-httpcomponents-client-5-api` to bom

### PR DESCRIPTION
**Pull request by strangelookingnerd**
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: sample-plugin/pom.xml
Comment (MarkEWaite) [inline sample-plugin/pom.xml:71]: I believe that this should be in the `test` scope as the dependencies that precede it are.  However, I need advice from those who are more experienced with the sample-plugin pom file, since `apache-httpcomponents-client-4-api` is only included in the `bom-weekly` pom and is not included in the sample plugin.

@jglick can you offer guidance on whether this new component should be in the `test` scope in the sample plugin or if it should only be in the `bom-weekly` pom like `apache-httpcomponents-client-4-api` ?
```suggestion
            <artifactId>apache-httpcomponents-client-5-api</artifactId>
            <scope>test</scope>
```

**Review by MarkEWaite — COMMENTED** [review]
Thanks for the pull request.  I have a question in the review comments that needs help from others before this can be merged.

For future pull requests, it is a little easier for reviewers and for you as a contributor if the pull request comes from a branch that is not named `master`.  I believe that the rationale for that is included in this [GitHub.com document](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) where it says:

> These changes are proposed in a branch, which ensures that the default branch only contains finished and approved work.

There is a longer explanation with more details of some of the risks in this [blog post](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/).

I don't think that you need to change this pull request to come from a different branch, but it would be nice if future pull requests came from a branch that is not the default branch (`master` in this repository).  Thanks again for the pull request and for being a  maintainer.

File: sample-plugin/pom.xml
Comment (jglick) [reply sample-plugin/pom.xml:71]: It can be added to `test` scope. However better still it is not add it at all and have it be picked up as a transitive dependency of some feature plugin that uses it and which is also in the BOM. Is there one?

Comment (strangelookingnerd) [reply sample-plugin/pom.xml:71]: I'm currently in the process of creating PR to plugins that may pick it up. Currently the only usage is in the upcoming release of [jenkinsci/custom-folder-icon-plugin](https://github.com/jenkinsci/custom-folder-icon-plugin/) where it is a `test` dependency.
So...shall I remove it from the sample-plugin?

**Review by jglick — APPROVED** [review]

**Comment by timja** [comment]
I assume it wasn't setup as a required check at the time?

**Pull request merged**
